src/lib.rs

```diff
@@ -0,0 +1,494 @@
+            if let Some(pod_spec) = pod_spec {
+                return match validate_environment_variables(&pod_spec, &validation_request.settings)
+                {
+                    Ok(_) => kubewarden::accept_request(),
+                    Err(err) => kubewarden::reject_request(Some(err.to_string()), None, None, None),
+                };
+            };
+            // If there is not pod spec, just accept it. There is no data to be
+            // validated.
+            kubewarden::accept_request()
+        }
+        Err(_) => kubewarden::reject_request(
+            Some("Cannot parse validation request".to_string()),
+            None,
+            None,
+            None,
+        ),
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use std::collections::HashSet;
+
+    #[test]
+    fn allareused_operator_ensure_all_envvar_are_defined_in_the_rule() -> Result<(), ()> {
+        let rule = Rule {
+            reject: settings::Operator::AllAreUsed,
+            environment_variables: HashSet::from([
+                settings::EnvVar {
+                    name: "name".to_string(),
+                    value: Some("value".to_string()),
+                },
+                settings::EnvVar {
+                    name: "name3".to_string(),
+                    value: Some("value3".to_string()),
+                },
+            ]),
+        };
+        let envvar = vec![
+            apicore::EnvVar {
+                name: "name".to_string(),
+                value: Some("value".to_string()),
+                ..Default::default()
+            },
+            apicore::EnvVar {
+                name: "name2".to_string(),
+                value: Some("value2".to_string()),
+                ..Default::default()
+            },
+        ];
+        let result = validate_envvar_with_rule(&rule, &envvar);
+        assert!(result.is_ok());
+        Ok(())
+    }
+
+    #[test]
+    fn allareused_should_reject_if_envvar_is_not_in_the_resource() -> Result<(), ()> {
+        let rule = Rule {
+            reject: settings::Operator::AllAreUsed,
+            environment_variables: HashSet::from([
+                settings::EnvVar {
+                    name: "name2".to_string(),
+                    value: Some("value2".to_string()),
+                },
+                settings::EnvVar {
+                    name: "name3".to_string(),
+                    value: Some("value3".to_string()),
+                },
+            ]),
+        };
+        let envvar = vec![
+            apicore::EnvVar {
+                name: "name2".to_string(),
+                value: Some("value2".to_string()),
+                ..Default::default()
+            },
+            apicore::EnvVar {
+                name: "name3".to_string(),
+                value: Some("value3".to_string()),
+                ..Default::default()
+            },
+        ];
+        let result = validate_envvar_with_rule(&rule, &envvar);
+        assert!(result.is_err());
```

Comment:
I would compare the returned `result.message`, to check we are verifying the correct case (same in the other test cases).